### PR TITLE
HPCC-27377 Add min & max skew fields to WsDfu.DFUQuery

### PR DESCRIPTION
**Pull request by wangkx**
1. Report pre-existing calculated skew values in WsDfu.DFUQuery.
2. Sort/filter logical files based on those skew values.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/services/ws_dfu/ws_dfuService.cpp [hunk 1]
```diff
@@ -3617,24 +3617,30 @@ void CWsDfuEx::setDFUQueryFilters(IEspDFUQueryRequest& req, StringBuffer& filter
...
+void CWsDfuEx::setInt64RangeFilter(__int64 from, __int64 to, DFUQFilterField filterID, StringBuffer &filterBuf)
+{
+    if ((from == 0) && (to == 0))
+        return;
+
+    filterBuf.append(DFUQFTinteger64Range).append(DFUQFilterSeparator);
+    filterBuf.append(getDFUQFilterFieldName(filterID)).append(DFUQFilterSeparator);
+    StringBuffer buf;
+    if (from > 0)
+        filterBuf.append(from);
```
Comment (jakesmith) [inline esp/services/ws_dfu/ws_dfuService.cpp:3637]: question: I assume it is correct/ok, and not new, but is it ok when from/to are missing, that it means the filterBuf will container multiple DFUQFilterSeparator without any value between them?
Would it be better to populate with a low value (0) for from, and a high value (max-int) for to ?

**Review by jakesmith — COMMENTED** [review]
@wangkx - 1 minor question

**Comment by jakesmith** [comment]
The regression suite failure is unrelated (workflow_9c.ecl in roxie)
@AttilaVamos - there probably a JIRA already for this sporadic failure?

File: esp/services/ws_dfu/ws_dfuService.cpp (hunk 1 above)
Comment (wangkx) [reply esp/services/ws_dfu/ws_dfuService.cpp:3637]: In CIterateFileFilterContainer (dadfs.cpp), there is the code to handle the multiple DFUQFilterSeparator without any value between them. The code uses the hasFilter and hasFilterHigh in checkInteger64RangeFilter() and checkIntegerRangeFilter().

**Comment by wangkx** [comment]
@jakesmith I added more changes. Please review. Thanks.

**Review by jakesmith — COMMENTED** [review]
@wangkx - looks good, please squash.

**Comment by wangkx** [comment]
squashed

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good

**Pull request merged**